pkgs/top-level/all-packages.nix

```
@@ -606,11 +606,20 @@ in
   makeWrapper = makeSetupHook { deps = [ dieHook ]; substitutions = { shell = targetPackages.runtimeShell; }; }
                               ../build-support/setup-hooks/make-wrapper.sh;
 
-  makeBinaryWrapper = makeSetupHook {
-    deps = [ dieHook ];
-  } (runCommand "make-binary-wrapper.sh" {} ''
-      substitute ${../build-support/setup-hooks/make-binary-wrapper.sh} $out --replace " cc " " ${gcc}/bin/cc "
-  '');
+  makeBinaryWrapper = let
+    f = { cc, sanitizers }: let
+      san = lib.concatMapStringsSep " " (s: "-fsanitize=${s}") sanitizers;
+      script = runCommand "make-binary-wrapper.sh" {} ''
+        substitute ${../build-support/setup-hooks/make-binary-wrapper.sh} $out \
+          --replace " @CC@ " " ${cc}/bin/cc ${san} "
+      '';
+    in
+      makeSetupHook { deps = [ dieHook ]; } script;
+  in
+    lib.makeOverridable f {
+      cc = gcc;
```

```suggestion
      cc = stdenv.cc.cc;
```
defaulting to gcc might cause greatly inflated build closures on darwin

---

I'm getting this error on my intel macbook pro (2019) after this change:
```sh
$ env NIX_PATH=nixpkgs=$PWD nix-shell -p makeBinaryWrapper --run "makeWrapper /usr/local/bin/python3 a"
bash: /nix/store/5im8ywgl0cilxlysvrwdnhik954nivbm-clang-7.1.0/bin/cc: No such file or directory
```

Seems like `cc` doesn't exist here:
```
$ ls /nix/store/5im8ywgl0cilxlysvrwdnhik954nivbm-clang-7.1.0/bin/
clang			  clang-check	      clang-import-test      clang-rename	   diagtool
clang++			  clang-cl	      clang-include-fixer    clang-reorder-fields  find-all-symbols
clang-7			  clang-cpp	      clang-offload-bundler  clang-tidy		   hmaptool
clang-apply-replacements  clang-format	      clang-query	     clangd		   modularize
clang-change-namespace	  clang-func-mapping  clang-refactor	     cpp		   scan-build
```

---

huh, didn't realize clang doesn't come with `cc` links. try `stdenv.cc`, that will *definitely* have a `cc`.

---

That seems to have worked, but now I get a new error:
```sh
env NIX_PATH=nixpkgs=$PWD nix-shell -p makeBinaryWrapper --run "makeWrapper /usr/local/bin/python3 a"
In file included from <built-in>:362:
<command line>:1:9: error: '_FORTIFY_SOURCE' macro redefined [-Werror,-Wmacro-redefined]
#define _FORTIFY_SOURCE 2
        ^
<built-in>:350:9: note: previous definition is here
#define _FORTIFY_SOURCE 0
        ^
1 error generated.
```

---

If I remove the sanitizers, the error goes away - so I guess it has something to do with those

---

Seems to be caused by the address sanitizer